src/modules/35-connectors/components/CreateConnector/GithubConnector/StepAuth/StepGithubAuthentication.tsx

```diff
@@ -175,13 +175,19 @@ const StepGithubAuthentication: React.FC<StepProps<StepGithubAuthenticationProps
     ]
 
     useEffect(() => {
-      if (props.isEditMode && props.connectorInfo && accountId) {
-        setupGithubFormData(props.connectorInfo, accountId).then(data => {
-          setInitialValues(data as GithubFormInterface)
-          setLoadingConnectorSecrets(false)
-        })
+      if (loadingConnectorSecrets) {
+        if (props.isEditMode) {
```

Comment:
Might as well combine these two
```suggestion
      if (loadingConnectorSecrets && props.isEditMode) {
```